pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_drop_column.go

```diff
@@ -246,6 +247,12 @@ func dropColumn(
 				undroppedSeqBackrefsToCheck.Add(e.SequenceID)
 			}
 		case *scpb.FunctionBody:
+			// Until the appropriate version gate is hit, we still do not allow
+			// dropping function bodies.
+			if !b.ClusterSettings().Version.IsActive(b, clusterversion.V23_1) {
```

Comment:
Unfortunately this is not enough for function because function elements are skipped before `v23_1` during decomposition. And funny that `Owner` element is not version gated, so I ran into some mixed version test error in #97631, so I added [this version gate ](https://github.com/cockroachdb/cockroach/pull/97631/files#diff-21dd460f3a39bc0cf0a7b11db9436f38bc6ac3c3e415e154b1e87cd11689c74fR120)to fallback all function relevant things until 23.1.